tests/phpunit/tests/cron/setCronArray.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * Test the `_set_cron_array()` function.
+ *
+ * @group cron
+ * @covers ::_set_cron_array
+ */
+class Tests_Cron_setCronArray extends WP_UnitTestCase {
+
+	public function set_up() {
+		parent::set_up();
+		// Make sure the schedule is clear.
+		_set_cron_array( array() );
+	}
+
+	public function tear_down() {
+		// Make sure the schedule is clear.
+		_set_cron_array( array() );
+		parent::tear_down();
+	}
+
+	/**
+	 * Tests the input validation for the `_set_cron_array()` function.
+	 *
+	 * Includes verifying that invalid input - typically `false` - does not result in a PHP
+	 * deprecation warning on PHP 8.1 or higher.
+	 *
+	 * The warning that we should not see:
+	 * `Deprecated: Automatic conversion of false to array is deprecated`.
+	 *
+	 * @ticket 53635
+	 *
+	 * @dataProvider data_set_cron_array_input_validation
+	 *
+	 * @param mixed $input    Cron "array".
+	 * @param array $expected Expected array entry count of the cron option after update.
+	 */
+	public function test_set_cron_array_input_validation( $input, $expected ) {
+		delete_option( 'cron' );
+		$this->assertTrue( _set_cron_array( $input ) );
+
+		$crons = get_option( 'cron' );
+		$this->assertIsArray( $crons, 'Cron option is not an array' );
+		$this->assertArrayHasKey( 'version', $crons, 'Cron option does not have a "version" key' );
+		$this->assertCount( $expected, $crons, 'Cron option does not contain the expected nr of entries.' );
```

Comment:
The full-stop (`.`) in this error message is inconsistent with the others in this test.

Reply:
Force of habit from comments, though I don't think it's a bad habit, so I've updated the other messages now to also contain punctuation.